sleap/gui/suggestions.py

```diff
@@ -291,6 +293,86 @@ def _velocity_video(
 
         return cls.idx_list_to_frame_list(frame_idxs, video)
 
+    @classmethod
+    def max_point_displacement(
+        cls,
+        labels: "Labels",
+        videos: List[Video],
+        displacement_threshold: float,
+        **kwargs,
+    ):
+        """Finds frames with maximum point displacement above a threshold."""
+
+        proposed_suggestions = []
+        for video in videos:
+            proposed_suggestions.extend(
+                cls._max_point_displacement_video(video, labels, displacement_threshold)
+            )
+
+        suggestions = VideoFrameSuggestions.filter_unique_suggestions(
+            labels, videos, proposed_suggestions
+        )
+
+        return suggestions
+
+    @classmethod
+    def _max_point_displacement_video(
+        cls, video: Video, labels: "Labels", displacement_threshold: float
+    ):
+        # ONCE labels.numpy works: delete lfs ~322 - 328
+        lfs = labels.find(video)
+        frames = len(lfs)
+
+        if frames < 2:
+            return []
+        
+
+        video_instances = labels.numpy(video=video, all_frames=True, untracked=False)
+        frames = len(video_instances)
+
+        if frames < 2:
+            return []
+
+        # ONCE labels.numpy works: delete print statements ~336 - 340
+        print('type of video_instances: ', type(video_instances))
+        print(video_instances[0])
+        print('type of video_instances[0]: ', type(video_instances[0]))
+        print(f"Number of elements returned by labels.numpy(): {video_instances.shape}")
+        print(f"Number of elements returned by labels.numpy(): {len(video_instances)}")
+
+
+        print('type of video_instances: ', type(video_instances))
+        print('type of video_instances[0]: ', type(video_instances[0]))
+
+
+        displacements = []
+        for idx in range(1, frames):
+            prev_points = video_instances[idx-1]
+            curr_points = video_instances[idx]
+
+
+            if prev_points.shape != curr_points.shape:
+                continue
+            
+            # Mask to identify non-nan values
+            valid_mask = ~np.isnan(prev_points) & ~np.isnan(curr_points)
+            # Filter out nan values
+            valid_prev_points = prev_points[valid_mask].reshape(-1, 2)
+            valid_curr_points = curr_points[valid_mask].reshape(-1, 2)
+
+            if valid_prev_points.size == 0 or valid_curr_points.size == 0:
+                continue
+
+            displacement = np.linalg.norm(valid_curr_points - valid_prev_points, axis=1).sum()
+            displacements.append((displacement, idx))
+
+        frame_idxs = [
+            frame_idx for displacement, frame_idx in displacements if displacement > displacement_threshold
+        ]
```

Comment:
The suggested approach handles nan values as we want by:
1. resulting in nan in the euclidean norm
2. Being excluded in the mean calculation for all points in an `Instance`

```suggestion
        # Get numpy of shape (frames, tracks, nodes, x, y)
        labels_numpy = labels.numpy(video=video, all_frames=True, untracked=False)

        # Return empty list if not enough frames
        n_frames, n_tracks, n_nodes, _ = labels_numpy.shape
        if n_frames < 2:
            return []

        # Calculate displacements
        diff = labels_numpy[1:] - labels_numpy[:-1]  # (frames - 1, tracks, nodes, x, y)
        euc_norm = np.linalg.norm(diff, axis=-1)  # (frames - 1, tracks, nodes)
        mean_euc_norm = np.nanmean(euc_norm, axis=-1)  # (frames - 1, tracks)

        # Find frames where mean displacement is above threshold
        threshold_mask = np.any(
            mean_euc_norm > displacement_threshold, axis=-1
        )  # (frames - 1,)
        frame_idxs = list(np.argwhere(threshold_mask).flatten())  # [0, len(frames - 1)]
```